agent/lib/kontena/workers/container_log_worker.rb

```diff
@@ -67,7 +68,7 @@ def on_message(id, stream, chunk)
         type: stream,
         data: data
       }
-      @queue << msg
+      @queue << msg if @queue.size < QUEUE_LIMIT
```

Comment:
Would it make sense to move the limit check to beginning of the method to avoid any parsing and matching if the queue is full. The agent has hands full at that point anyway so no need to do unnecessary parsing.